Chessington.GameEngine/Pieces/Bishop.cs

```diff
@@ -10,7 +11,90 @@ public Bishop(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            var moves = Enumerable.Empty<Square>();
+            var piece = board.FindPiece(this);
+
+            for (var i = 1; i < 8; i++)
+            {
+                var row = piece.Row + i;
+                var col = piece.Col + i;
+                if (row >= 0 && row <= 7 && col >= 0 && col <= 7)
+                {
+                    if (board.GetPiece(new Square(row, col)) != null)
+                    {
+                        if (board.GetPiece(new Square(row, col)).Player != board.GetPiece(piece).Player)
+                        {
+                            moves = moves.Concat(new[] { new Square(row, col) });
+                        }
+
+                        break;
+                    }
+
+                    moves = moves.Concat(new[] { new Square(row, col) });
+                }
+            }
+
+            for (var i = 1; i < 8; i++)
+            {
+                var row = piece.Row - i;
+                var col = piece.Col - i;
+                if (row >= 0 && row <= 7 && col >= 0 && col <= 7)
```

Comment:
I think the logic inside the for loop can be commonized, since the only difference is row/column